src/sources/host_metrics/cgroups.rs

```diff
@@ -0,0 +1,271 @@
+use super::{filter_result_sync, FilterList, HostMetricsConfig};
+use crate::event::metric::Metric;
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use shared::btreemap;
+use std::fs::{self, File};
+use std::io::{self, Read};
```

Comment:
By using `std::fs` and `std::io` we're missing async and sync code. The caller will be the async source and its dependencies ought to be async too. For the most part it won't matter except in those cases where the kernel is delayed in handing us back a file handler for cgroup stuff and the source sort mysteriously pauses for a second.

Reply:
Can this delay actually ever happen? That is, is it worth having the constant overhead, admittedly small, of going through the async runtime in case the kernel will stall out one of the steps, when that stall is vanishingly rare in real operation (beyond the normal scheduling micro-delays)? I am open to rewriting it to full async code and we do _indirectly_ pull in `async-std` so it should not be a difficult task at all, it just seems like a waste of code.

Reply:
> Can this delay actually ever happen? 

Sure. Consider what happens when we're close to the file descriptor limit or are running under a profiler that instruments file opens. In the first case we may see open syscalls appreciably delayed and in the later we will see them slow depending on the intercept. This later is maybe more concerning because it'll mis-instrument vector when running under a profiler. 

With regard to the potential overhead of going through the runtime versus profiling if that's a concern here I'd be happy to reconsider if there are benchmarks to show that overhead's significant. I'm more concerned with consistency of behavior. I have a strong suspicion that the stalling we see in vector in practice -- where capacity drops off a cliff and then picks back up -- is down to the mixing of sync and async code. Very hard to actually debug/diagnose and we can really only catch it when vector's all-up. If we push to make things as async as possible that inconsistency in behavior goes away, is my thinking.

> I am open to rewriting it to full async code and we do indirectly pull in async-std so it should not be a difficult task at all, it just seems like a waste of code.

Why would we use `async-std`? Tokio has filesystem handling code.

Reply:
Got it. I'll do that.

Reply:
Resolved in f2a7eee42